pennylane/debugging/debugger.py

```diff
@@ -132,33 +132,198 @@ def pldb_device_manager(device):
 
 
 def breakpoint():
-    """Launch the custom PennyLane debugger."""
+    """A function which freezes execution and launches the PennyLane debugger (PLDB).
+
+    This function marks a location in a quantum circuit (QNode). When it is encountered during
+    execution of the quantum circuit, an interactive debugging prompt is launched to step
+    throught the circuit execution using Pdb like commands (:code:`list`, :code:`next`,
+    :code:`continue`, :code:`quit`).
+
+    **Example**
+
+    Consider the following python script containing the quantum circuit with breakpoints.
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.breakpoint()
+
+            qml.RX(x, wires=0)
+            qml.Hadamard(wires=1)
+
+            qml.breakpoint()
+
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.Z(0))
+
+        circuit(1.23)
+
+    Running the above python script opens up the interactive :code:`[pldb]:` prompt in the terminal.
```

Comment:
Outside of PL debugging, is it normally `[pdb]:`?

Reply:
Its `(Pdb) `